src/TaskProviderDune.ml

```diff
@@ -0,0 +1,104 @@
+open Import
+
+let dune_bin = "dune"
+
+let task_type = "dune"
+
+let taskDefinition = { Task.type_ = task_type }
+
+let source = task_type
+
+let problemMatchers = [| "ocamlc" |]
```

Comment:
Shouldn't this be `$ocamlc`? [The docs](https://code.visualstudio.com/api/references/vscode-api#Task) use '$tsc' and '$eslint' as examples.

Reply:
https://github.com/Khady/vscode-ocaml-platform/blob/ef120ec9810dec8ec222c689e16873e8990473aa/package.json#L80

The name must be the one declared by the extension. Also it seems that the names starting with a $ are supposed to be reserved for extensions that are part of vscode.

Reply:
> Also it seems that the names starting with a $ are supposed to be reserved for extensions that are part of vscode.

I don't think so. When I use the task provider as-is, I get no problems reported, but it works fine if I add the dollar sign.

 Also if I do `Tasks: Configure Task` for `"dune: build \dune-project`, it adds the following to my tasks.json:

```json
{
	"type": "dune",
	"problemMatcher": [
		"ocamlc"
	],
	"group": "build",
	"label": "dune: build \\dune-project"
}
```

But there is an error:
> Value is not accepted. Valid values: "$msCompile", "$lessCompile", "$gulp-tsc", "$jshint", "$jshint-stylish", "$eslint-compact", "$eslint-stylish", "$go", "$lessc", "$node-sass", "$tsc", "$tsc-watch", "**$ocamlc**".